playbook/app/pb_kits/playbook/pb_timestamp/_timestamp.html.erb

```diff
@@ -1,6 +1,16 @@
 <%= content_tag(:div,
-    id: object.id,
-    data: object.data,
-    class: object.classname) do %>
-  <%= pb_rails("caption", props: { text: object.text, tag: 'span', size: 'xs' }) %>
+  aria: object.aria,
+  id: object.id,
+  data: object.data,
+  class: object.classname) do %>
+
+  <% if object.variant == "updated" || object.variant == "elapsed" %>
+    <%= pb_rails("caption", props: { text: object.format_updated_string, size: 'xs', dark: object.dark }) %>
+  <% else %>
+    <% if object.show_date %>
+      <%= pb_rails("caption", props: { text: object.timestamp ? object.format_datetime_string : object.text, size: 'xs', dark: object.dark }) %>
```

Comment:
Ternaries should probably be abstracted to a method. Think of a method name that suits this logic and move this part.